### PR TITLE
Add post-commit hooks support

### PR DESCRIPTION
**Pull request by xobb1t**
Main goal is to add support of github post-commit web hooks, that will start lint process.

### CODE REVIEW

**Comment by xobb1t** [comment]
I have one question: do we need to split webhooks app to hooks and commits apps?

**Comment by yumike** [comment]
It's up to you. I think, it is not very important, because we are moving to Flask in the near future.

**Comment by xobb1t** [comment]
I'm done. @yumike, take a look, please.

File: project/lint/tasks.py [hunk 1]
```diff
@@ -23,23 +23,38 @@ def save_result(report, result):
...
+                report.save()
     decorator.__name__ = func.__name__
     return decorator
 
 
 @task()
 @exception_handle
-def process_report(report):
+def process_report(report_pk=None, commit_pk=None):
+    if commit_pk is None and report_pk is None:
```
Comment (xobb1t) [inline project/lint/tasks.py]: This part looks very complicated, don't know, may be better will be to move it outside of task?

File: project/lint/ghclone.py [hunk 2]
```diff
@@ -32,9 +32,9 @@ def tempdir(root=None):
 
 
 @contextmanager
-def clone(url):
+def clone(url, head=None):
```
Comment (yumike) [inline project/lint/ghclone.py]: `head` is actually not a head, but a `ref` (reference to a commit), if I understand you correctly.

Comment (yumike) [inline project/lint/ghclone.py]: Sorry, not a `ref`, but a `hash` :)

Comment (xobb1t) [inline project/lint/ghclone.py]: Ok, will rename it.

File: project/lint/tasks.py (hunk 1 above)
Comment (yumike) [inline project/lint/tasks.py]: You can just move report creation by commit to another task (which will add `process_report` task to the queue in the end) or to the `webhooks.views.handler` view.

Comment (xobb1t) [inline project/lint/tasks.py]: Great advice, thanks!

**Comment by xobb1t** [comment]
Looks better after the `process_report` has been splitted to `process_commit` and `process_report`.

**Comment by yumike** [comment]
Merge :thumbsup:

**Pull request merged**